custom-metrics-stackdriver-adapter/Dockerfile

```diff
@@ -12,10 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.13-alpine as builder
+FROM golang:1.16-alpine as builder
 WORKDIR ${GOPATH}/src/github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter
 COPY . ./
-RUN CGO_ENABLED=0 GOOS=linux go build -mod=vendor -o /adapter
+RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly -o /adapter
```

Comment:
I don't think we should switch away from vendor here.
The vendor directory enables us to have reproducible and faster builds as it's avoiding pulling dependencies on every build.
See https://golang.org/ref/mod#build-commands for more details.

In this case, all readonly does is prevent the build process from changing go.mod/go.sum, but if a dependency is temporary or permanently unavailable the build will still fail.

To update the vendor directory one should run
```sh
go mod tidy
go mod vendor
```

Reply:
> The vendor directory enables us to have reproducible and faster builds as it's avoiding pulling dependencies on every build.

We have deterministic build behaviour with the help of go.sum file. It's faster for the first time for the local build.

> if a dependency is temporary or permanently unavailable

Go module proxy caches these packages (public packages) permanently. For details, search for `I removed a bad release from my repository but it still appears in the mirror, what should I do?` in https://proxy.golang.org/

A dependency package might be temporarily unavailable, I don't see it as a risk, especially for automated build.

One potential benefit of using vendor is that we could see the diff between third party packages update. I believe we can get these information from commit log/release notes.